providers/default/defltprov.c

```diff
@@ -121,19 +121,18 @@ static const OSSL_ALGORITHM deflt_ciphers[] = {
     { "AES-192-OCB", "default=yes", aes192ocb_functions },
     { "AES-128-OCB", "default=yes", aes128ocb_functions },
 #endif /* OPENSSL_NO_OCB */
-/* TODO(3.0) Add aliases when they are supported */
-    { "id-aes256-GCM", "default=yes", aes256gcm_functions },
-    { "id-aes192-GCM", "default=yes", aes192gcm_functions },
-    { "id-aes128-GCM", "default=yes", aes128gcm_functions },
-    { "id-aes256-CCM", "default=yes", aes256ccm_functions },
-    { "id-aes192-CCM", "default=yes", aes192ccm_functions },
-    { "id-aes128-CCM", "default=yes", aes128ccm_functions },
-    { "id-aes256-wrap", "default=yes", aes256wrap_functions },
-    { "id-aes192-wrap", "default=yes", aes192wrap_functions },
-    { "id-aes128-wrap", "default=yes", aes128wrap_functions },
-    { "id-aes256-wrap-pad", "default=yes", aes256wrappad_functions },
-    { "id-aes192-wrap-pad", "default=yes", aes192wrappad_functions },
-    { "id-aes128-wrap-pad", "default=yes", aes128wrappad_functions },
+    { "id-aes256-GCM:AES-256-GCM", "fips=yes", aes256gcm_functions },
+    { "id-aes192-GCM:AES-192-GCM", "fips=yes", aes192gcm_functions },
+    { "id-aes128-GCM:AES-128-GCM", "fips=yes", aes128gcm_functions },
+    { "id-aes256-CCM:AES-256-CCM", "fips=yes", aes256ccm_functions },
+    { "id-aes192-CCM:AES-192-CCM", "fips=yes", aes192ccm_functions },
+    { "id-aes128-CCM:AES-128-CCM", "fips=yes", aes128ccm_functions },
+    { "id-aes256-wrap:AES-256-WRAP:AES256-WRAP", "fips=yes", aes256wrap_functions },
```

Comment:
Why do we allow "AES-256" and "AES256" here - but not with other algorithms?

Reply:
That's a question I cannot answer...  I know that these names exist in previous OpenSSL versions:

```
$ openssl list -cipher-algorithms | grep -i aes | grep -i wrap
aes128-wrap => id-aes128-wrap
aes192-wrap => id-aes192-wrap
aes256-wrap => id-aes256-wrap
id-aes128-wrap
id-aes128-wrap-pad
id-aes192-wrap
id-aes192-wrap-pad
id-aes256-wrap
id-aes256-wrap-pad
```

The `id-` names are from the ASN.1 module, and `aesNNN-wrap` is our "friendlier" name that only half compliant with our other names, so I added `aes-NNN-wrap` to be aligned with our usual *cipher*-*bits*-*mode* naming.

Reply:
So basically, `AES256` is the odd name that I haven't replicated because it's relatively uncommon on our naming.

Is this a matter that should be resolved here, or in a separate naming PR?

Reply:
So the "AES256-WRAP" name is newly introduced in master? If so, I would drop it in favour of "AES-256-WRAP" so we can have a consistent format for all "friendly" names (and I would put that name first).

Reply:
No, "AES-256-WRAP" is newly introduced in this PR.

I forgot a detail:

```
: ; openssl version
OpenSSL 1.1.1d  10 Sep 2019
```